build.sbt

```diff
@@ -34,6 +34,7 @@ def mimaSettings: Seq[Setting[_]] = Seq(
   },
 )
 
+Global / onChangedBuildSource := ReloadOnSourceChanges
```

Comment:
Could we not add this, please?  It's a departure from the default behaviour, which I prefer.

Reply:
I agree that this setting is ideally either in `~/.sbt/global.sbt` or in a gitignored local file (I added `local.sbt` to my global gitignore for things like this).

Reply:
I've been on a vim + `~` kick lately, and with `~` `ReloadOnSourceChanges` is really convenient. If you guys think it should be moved to `local.sbt` I'll do it, but shouldn't we be dogfooding all the knobs so we'll catch when it bugs out?

Reply:
Is there a way that we could make the value in `~/.sbt/global.sbt` always win?

I do think that for many, but not all builds, auto reload is a much better default. I leave it off in sbt itself though. The reason is say that I bump a library version that is backward compatible with the old version, I then rebase to an earlier commit that was originally written against the old library api, I really don't want to reload both because reload itself takes a while in sbt (like 5 seconds or more -- much of which I now know is spent building the compiled setting map: https://github.com/sbt/sbt/pull/5666) and because changing the library might end up effectively invalidating the entire sbt project since zinc doesn't do api hashing of jars on the classpath and invalidates everything if the jar changes. If reload were fast and zinc were better (or configured differently) about invalidating binary dependencies, I think auto reload would be the no-brainer default.

Reply:
I also agree with your point about dogfooding. https://github.com/sbt/sbt/pull/5673/commits/366c49a7641de33e5520f8d0ec0898e0bb19b99e was a direct result of my dogfooding the vim + `~` workflow in a new project I'm working on.

Reply:
Here's my proposal related to this - https://github.com/sbt/sbt/issues/5679

Reply:
https://github.com/sbt/zinc/pull/829/commits/8de29c50bd3a2e981092371c2b1972731723c05e no longer contains `Global / onChangedBuildSource`

Reply:
I would love to not have to think about whether I need to rebuild the state or not, but while it takes the time it does and how it affects zinc, I prefer to stay in control.

I'm really excited about the compiled setting map caching, hoping it doesn't hit any bugs.  I also built https://github.com/dwijnand/sbt-reloadquick/ way back when, so I could skip some waiting - perhaps there's something there that can be folded into `reload`.